### PR TITLE
Implement Serializers for Ed25519 and Ed448

### PR DESCRIPTION
**Pull request by mattcaswell**
This adds the serializers and associated tests for the Ed25519 and Ed448 algorithms. The docs were already done when the X25519/X448 serializers were added since the interface is the same.

This is WIP  because it depends on and includes the commits from #11261. Aside from that it is complete.

### CODE REVIEW

**Comment by slontis** [comment]
Looks like a fairly simple change in the last 2 commits. (tests need some fixing - but it looks almost there)

**Comment by mattcaswell** [comment]
Now that #11261 has gone in I've rebased this, and fixed the travis issues.

This is now ready to review - please take a look.

**Comment by slontis** [comment]
1 travis error left - probably related to no-ec

**Comment by mattcaswell** [comment]
Fixup pushed to address  the travis failure.

**Comment by paulidale** [comment]
A quick push of this after the defrosting of the repo would be nice.
The keygen is semi-dependent.

**Comment by slontis** [comment]
will try merging this again now..

**Comment by slontis** [comment]
Merged to master.

**Comment by mattcaswell** [comment]
Thanks for  merging this @slontis

File: crypto/ec/ecx_meth.c [hunk 1]
```diff
@@ -31,6 +31,11 @@
 #define KEYLENID(id)    (IS25519(id) ? X25519_KEYLEN \
                                      : ((id) == EVP_PKEY_X448 ? X448_KEYLEN \
                                                               : ED448_KEYLEN))
+#define KEYNID2TYPE(id) \
+    (IS25519(id) ?  ECX_KEY_TYPE_X25519 \
+                 : ((id) == EVP_PKEY_X448 ? ECX_KEY_TYPE_X448 \
+                                          : ((id) == EVP_PKEY_ED25519 ? ECX_KEY_TYPE_ED25519 \
```
Comment (kaduk) [inline crypto/ec/ecx_meth.c:37]: Isn't the first `IS25519` going to catch both x25519 and ed25519?  I'm not sure how this last `EVP_PKEY_ED25519` check is going to do anything.

**Pull request closed without merging**